articles/feature-engineering-in-machine-learning/index.md

```diff
@@ -0,0 +1,167 @@
+
+Features are extracted from raw data. These features are then transformed into formats compatible with the machine learning process. Domain knowledge of data is key to the process. Along with domain knowledge, both programming and math skills are required to perform feature engineering. The correct features make machine learning algorithms successful.
+
+This is one of the most effective ways to improve model performance. But what is a feature? According to Wikipedia, a [feature](https://en.wikipedia.org/wiki/Feature_engineering) is a property shared by all the independent units on which analysis is to be done.
+
+It is a characteristic that might assist in problem solving. As beneficial as this process is, it is often described as being time consuming. It also is a difficult process. Furthermore, it requires domain expertise for the data being analyzed. So, why exactly do we need this process?
+
+#### Need for Feature Engineering
+Features influence results of predictive models. I'd like to highlight two points on the importance of this process:
+
+1. **Reduced complexity**: Algorithms are fed raw data in order to build these models. However, these algorithms make predictions without a clear guide. Feature engineering guide these algorithms. But what is the benefit of this? For starters, with correct features, model complexity will reduce. Correct scope and purpose of the model makes the process more efficient. This makes it simpler to understand, build, modify, and maintain models.
+
+2. **Increased accuracy**: Feature engineering is a broad process. It involves transforming variables into suitable formats. It involves creation of new variables from existing ones. Filling in missing values, among other processes, is also under this umbrella. These processes ultimately influence accuracy of the model. When done correctly, with the correct data, it increases the accuracy of the models.
+
+#### Features and Techniques
+We have defined what a feature is. But, for a wholesome understanding, it is important that we delve deeper to understand what kinds of features exist. After that, we finally will go over a few feature engineering techniques.
+
+Below are three general types of features.
+
+##### Examples of Features
+- **Categorical features**. These are features derived from categorical data. Categorical data is data that is grouped into categories. It is often in a non-numerical format. An example of such data could be, a type of pet. This could be a dog, cat, snake, hamster and so forth.
+
+- **Text features**. Text features are derived from text data. This means that text is encoded to corresponding numerical values.
+
+- **Image features**. Images are converted into suitable formats for analysis. The result is image-based features.
+
+Understanding of features puts us in a better position to dive into feature engineering techniques.  
+
+Let’s briefly define 5 techniques.
+
+##### 5 Feature Engineering Techniques
```

Comment:
Since this is an article about feature engineering, more information about the mentioned techniques would aid in better understanding. For example, under imputation, I would mention how the unavailable values are filled. I can remove those rows or fill those values with the mean, median, etc., of the column. Similarly, more information about all the techniques mentioned would provide a lot of value to the reader.